cmd/proxy/main.go

```diff
@@ -3,28 +3,52 @@ package main
 import (
 	"flag"
 	"net/http"
+	"os"
+	"strconv"
+	"strings"
 
 	log "github.com/sirupsen/logrus"
 
 	"slime.io/slime/modules/lazyload/pkg/proxy"
 )
 
+const WormholePorts = "wormholePorts"
+
 func main() {
+	// start health check server
 	go func() {
 		handler := &proxy.HealthzProxy{}
-		log.Println("Starting health check on :8080")
-		if err := http.ListenAndServe(":8080", handler); err != nil {
+		log.Println("Starting health check on :18080")
+		if err := http.ListenAndServe(":18080", handler); err != nil {
 			log.Fatal("ListenAndServe:", err)
 		}
 	}()
 
-	addr := flag.String("addr", "0.0.0.0:80", "The addr of the application.")
-	flag.Parse()
+	wormholePorts := os.Getenv(WormholePorts)
+	whPorts := strings.Split(wormholePorts, ";")
+
+	// start multi ports defined in WormholePorts
+	for idx, whPort := range whPorts {
 
-	handler := &proxy.Proxy{}
+		if whPort == "18080" {
+			log.Errorf("wormholePort can not be 18080, which is reserved for health check")
+			os.Exit(1)
+		}
 
-	log.Println("Starting proxy server on", *addr)
-	if err := http.ListenAndServe(*addr, handler); err != nil {
-		log.Fatal("ListenAndServe:", err)
+		p, err := strconv.Atoi(whPort)
+		if err != nil {
+			log.Errorf("wrong wormholePort value %s", whPort)
+			os.Exit(1)
+		}
+		handler := &proxy.Proxy{WormholePort: p}
+
+		addr := flag.String("addr"+strconv.Itoa(idx), "0.0.0.0:"+whPort, "The addr of the application.")
+		flag.Parse()
+
+		if idx != len(whPorts)-1 {
+			go handler.StartServer(addr)
```

Comment:
only the last server failure will cause the app to exit?

Reply:
change, main func will exit when all web servers are closed